heartbeat/scheduler/scheduler.go

```diff
@@ -21,22 +21,29 @@ import (
 	"errors"
 	"runtime/debug"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/elastic/beats/libbeat/logp"
 )
 
+const (
+	statePreRunning = int32(iota + 1)
+	stateRunning
+	stateDone
+)
+
 type Scheduler struct {
-	limit   uint
-	running bool
+	limit uint
+	state *int32
```

Comment:
I guess that's an atomic? Why do we need this to be a pointer?

Consider `running atomic.Bool`. The libbeat package provides some more typed atomics.

Reply:
I see you introduced 3 states. How about `atomic.Int`?

Reply:
++ on the atomic Int we have.

I used a pointer because the atomic operations in the stdlib require a pointer.